src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -364,6 +364,20 @@ void maybeAddPartitions(int minPartitionNum) throws ExecutionException, Interrup
       return newPartitionAssignments;
     }
 
+    private static BrokerMetadata randomBroker(Set<BrokerMetadata> brokers) {
+      int brokerSetSize = brokers.size();
+      // In practicality, the Random object should be rather more shared than this.
+      int random = new Random().nextInt(brokerSetSize);
+      int index = 0;
+      for (BrokerMetadata brokerMetadata : brokers) {
```

Comment:
Can we just change input to a list so that you don't need to create a index and iterate through this set?
If it is a list, you can just do list.get(random)

Reply:
sure

Reply:
I looked at the references to this variable (brokers) and I'd say let's keep the usage of Set as all the implementations of different methods in the class use the data structure interface `Set<E>`, especially for blacklistedBrokers. 

Changing to List would require changes to the Interface `public interface TopicFactory`

  /**
   * @param zkUrl zookeeper connection url
   * @return A set of brokers that don't take new partitions or reassigned partitions for topics.
   */
  Set<Integer> getBlackListedBrokers(String zkUrl);

Reply:
will address your concern in a separate PR